### PR TITLE
Simplify scheme

### PR DESCRIPTION
**Pull request by matteoettam09**

### CODE REVIEW

File: kill_and_cleanup.sh [hunk 1]
```diff
@@ -1,5 +1,16 @@
 #!/bin/bash
 
+# Check for processes using ports 8000 and 8001 and kill them if found
+for port in 8000 8001; do
```
Comment (mlegls) [inline kill_and_cleanup.sh:4]: should use lock file in /tmp/ or /var/ rather than hardcoded ports

File: src/lib.rs [hunk 2]
```diff
@@ -4,18 +4,19 @@ pub mod contracts;
...
+pub mod shared;
 
 /// A Python module implemented in Rust.
 #[pymodule]
 fn apiars(py: Python, m: &Bound<'_, PyModule>) -> PyResult<()> {
+    erc_for_job::add_erc_submodule(py, m)?;
     erc20_for_job::add_erc20_submodule(py, m)?;
     erc721_for_job::add_erc721_submodule(py, m)?;
     bundle_for_job::add_bundle_submodule(py, m)?;
-
```
Comment (mlegls) [inline src/lib.rs:18]: probable merge conflict with https://github.com/CoopHive/Apiary/commit/ed16c9ab341c79213d5477f9d61f5dc12f018e49

File: src/shared.rs [hunk 3]
```diff
@@ -35,35 +27,3 @@ pub fn py_val_err(msg: impl Into<String>) -> PyErr {
 pub fn py_run_err(msg: impl Into<String>) -> PyErr {
     PyErr::new::<PyRuntimeError, _>(msg.into())
 }
-
```
Comment (mlegls) [inline src/shared.rs:38]: probable merge conflict with https://github.com/CoopHive/Apiary/commit/d764d641c5f8a3a506e18c1c0099f80a18739783

File: kill_and_cleanup.sh (hunk 1 above)
Comment (matteoettam09) [reply kill_and_cleanup.sh]: https://github.com/CoopHive/Apiary/commit/e3a3918fe5feca0fad05baed485086f6cb5c2817

File: src/apiary/erc_for_job.rs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+use alloy::{
+    primitives::{self, Address, FixedBytes},
+    sol_types::SolValue,
+};
+
+use crate::shared::{py_val_err, py_run_err, ERC20Price, ERC721Price, BundlePrice};
+use crate::contracts::{ERC20PaymentObligation, JobResultObligation, IEAS};
+
+pub struct JobPayment20 {
+    pub price: ERC20Price,
```
[the remark this answers is not in the document]
Comment (matteoettam09) [reply src/apiary/erc_for_job.rs]: https://github.com/CoopHive/Apiary/commit/fc5f2ca8ea37f01ac67e095bb2b8bea361b89096

**Pull request merged**